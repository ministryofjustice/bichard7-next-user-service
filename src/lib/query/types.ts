export type DBResult<T> = T
export type TransformResult<T> = T
export type Transform<I, O> = (dbResult: DBResult<I>) => TransformResult<O>
export type QueryError = Error
export type OnError = (e: QueryError) => void

export enum QueryType {
  Any = "Any"
}
