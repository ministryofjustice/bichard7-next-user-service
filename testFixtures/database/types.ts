export enum Tables {
  Users = 'users'
}