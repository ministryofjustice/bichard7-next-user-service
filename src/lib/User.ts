export interface UserCredentials {
  emailAddress: string
  password: string
}

export type UserRole = "B7Allocator" | "B7ExceptionHandler"

export interface User {
  displayName: string
  emailAddress: string
  role: UserRole
}
