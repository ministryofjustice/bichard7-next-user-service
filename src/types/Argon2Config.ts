export default interface Argon2Config {
  hashLength: number
  memoryCost: number
  parallelism: number
  saltLength: number
  timeCost: number
}
