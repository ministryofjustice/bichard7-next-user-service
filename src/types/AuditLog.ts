import { randomUUID } from "crypto"
import KeyValuePair from "./KeyValuePair"

export default class AuditLog {
  public readonly auditLogId: string

  public readonly timestamp: Date

  constructor(
    public readonly action: string,
    public readonly username: string,
    public readonly userIp: string,
    public readonly requestUri: string,
    public readonly attributes?: KeyValuePair<string, unknown>
  ) {
    this.auditLogId = randomUUID()
    this.timestamp = new Date()
  }
}
