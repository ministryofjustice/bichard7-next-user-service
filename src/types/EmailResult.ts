export default interface EmailResult {
  subject: string
  body: string
}
