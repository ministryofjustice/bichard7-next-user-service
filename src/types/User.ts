import { UserGroupResult } from "./UserGroup"

interface User {
  id: number
  username: string
  exclusionList: string[]
  inclusionList: string[]
  endorsedBy: string
  orgServes: string
  forenames: string
  surname: string
  emailAddress: string
  groups: UserGroupResult[]
}

export default User
