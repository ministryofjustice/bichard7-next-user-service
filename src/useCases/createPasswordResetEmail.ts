import { generatePasswordResetToken, PasswordResetTokenPayload } from "lib/token/passwordResetToken"
import EmailResult from "types/EmailResult"
import { isError, Result } from "types/Result"
import User from "types/User"

export default (user: User, passwordResetCode: string): Result<EmailResult> => {
  const payload: PasswordResetTokenPayload = { emailAddress: user.emailAddress, passwordResetCode }
  const token = generatePasswordResetToken(payload)

  if (isError(token)) {
    return token
  }

  const url = new URL("/login/reset-password", "http://localhost:3000")
  url.searchParams.append("token", token)

  const subject = "Password reset request"
  const body = `
  Hi ${user.forenames} ${user.surname}
  Click here to reset your password:
  ${url.href}
  `
  return { subject, body }
}
