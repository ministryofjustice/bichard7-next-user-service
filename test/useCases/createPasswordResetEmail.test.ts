import EmailResult from "types/EmailResult"
import { isError } from "types/Result"
import User from "types/User"
import createPasswordResetEmail from "useCases/createPasswordResetEmail"

it("should generate the email subject and body", () => {
  const user = {
    username: "Dummy username",
    emailAddress: "Dummy email address",
    forenames: "Dummy forenames",
    surname: "Dummy surname"
  } as unknown as User

  const result = createPasswordResetEmail(user, "123")

  expect(isError(result)).toBe(false)

  const { subject, body } = result as EmailResult
  expect(subject).toMatchSnapshot()
  expect(body).toMatchSnapshot()
})
