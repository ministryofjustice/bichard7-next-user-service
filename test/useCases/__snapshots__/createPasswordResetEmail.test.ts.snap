// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate the email subject and body 1`] = `"Password reset request"`;

exports[`should generate the email subject and body 2`] = `
"
  Hi Dummy forenames Dummy surname
  Click here to reset your password:
  http://localhost:3000/login/reset-password?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbEFkZHJlc3MiOiJEdW1teSBlbWFpbCBhZGRyZXNzIiwicGFzc3dvcmRSZXNldENvZGUiOiIxMjMiLCJpYXQiOjE2Mjg4NzA4MjksImV4cCI6MTYyODg3MjYyOSwiaXNzIjoiQmljaGFyZCJ9.kVkTb0rfBHbAKUd2sb8Xevpj9WfIn1r7oGRlYITehC0
  "
`;
